### PR TITLE
Pass whitelist/blacklist via file path

### PR DESCRIPTION
**Pull request by jhs7jhs**
Deleted whitelist/blacklist option, and added whitelist-path/blacklist-path option passing the file path.
The file should contain IP addresses seperated with whitespaces.

### CODE REVIEW

File: codechain/config/mod.rs [hunk 1]
```diff
@@ -101,8 +102,30 @@ impl Config {
...
             self.network.bootstrap_addresses.iter().map(|s| SocketAddr::from_str(s).unwrap()).collect::<Vec<_>>();
-        let whitelist = self.network.whitelist.iter().map(|s| s.parse().unwrap()).collect::<Vec<_>>();
-        let blacklist = self.network.blacklist.iter().map(|s| s.parse().unwrap()).collect::<Vec<_>>();
+
+        let whitelist = self
+            .network
+            .whitelist_path
+            .as_ref()
+            .map(|path| {
+                let mut file = fs::File::open(path).expect("Whitelist file must exist");
```
Comment (sgkim126) [inline codechain/config/mod.rs]: Use [read_to_string](https://doc.rust-lang.org/std/fs/fn.read_to_string.html).
```rust
fs::read_to_string(path)
    .expect("Whitelist file must exist")
    .split_whitespace()
    .filter(|s| s.len() != 0)
    .map(|s| s.parse().unwrap())
    .collect()
```

File: codechain/config/mod.rs [hunk 2]
```diff
@@ -101,8 +102,30 @@ impl Config {
...
+                file.read_to_string(&mut content).expect("Failed to read the whitelist file");
+                content.split_whitespace().filter(|s| s.len() != 0).map(|s| s.parse().unwrap()).collect::<Vec<_>>()
+            })
+            .unwrap_or_else(Vec::new);
+        let blacklist = self
+            .network
+            .blacklist_path
+            .as_ref()
+            .map(|path| {
+                let mut file = fs::File::open(path).expect("Blacklist file must exist");
```
Comment (sgkim126) [inline codechain/config/mod.rs]: ditto.

Comment (jhs7jhs) [reply codechain/config/mod.rs]: Fixed it.

File: codechain/codechain.yml [hunk 3]
```diff
@@ -222,18 +222,16 @@ args:
         takes_value: true
         conflicts_with:
             - no-shard-validator
-    - whitelist:
-        long: whitelist
-        value_name: WHITELIST
-        help: Whitelisted IP addresses.
+    - whitelist-path:
+        long: whitelist-path
+        value_name: WHITELIST_PATH
```
Comment (kseo) [inline codechain/codechain.yml]: WHITELIST_PATH -> PATH

File: codechain/codechain.yml [hunk 4]
```diff
@@ -222,18 +222,16 @@ args:
...
+        help: Specify the path of a network whitelist file.
         takes_value: true
-        multiple: true
-    - blacklist:
-        long: blacklist
-        value_name: BLACKLIST
-        help: Blacklisted IP addresses.
+    - blacklist-path:
+        long: blacklist-path
+        value_name: BLACKLIST_PATH
```
Comment (kseo) [inline codechain/codechain.yml]: BLACKLIST_PATH -> PATH

File: codechain/codechain.yml [hunk 5]
```diff
@@ -222,18 +222,16 @@ args:
...
         conflicts_with:
             - no-shard-validator
-    - whitelist:
-        long: whitelist
-        value_name: WHITELIST
-        help: Whitelisted IP addresses.
+    - whitelist-path:
+        long: whitelist-path
+        value_name: WHITELIST_PATH
+        help: Specify the path of a network whitelist file.
```
Comment (kseo) [inline codechain/codechain.yml]: Specify the path for the network whitelist file.

File: codechain/codechain.yml [hunk 6]
```diff
@@ -222,18 +222,16 @@ args:
...
         takes_value: true
-        multiple: true
-    - blacklist:
-        long: blacklist
-        value_name: BLACKLIST
-        help: Blacklisted IP addresses.
+    - blacklist-path:
+        long: blacklist-path
+        value_name: BLACKLIST_PATH
+        help: Specify the path of a network blacklist file.
```
Comment (kseo) [inline codechain/codechain.yml]: Specify the path for the network blacklist file.

File: codechain/config/mod.rs [hunk 7]
```diff
@@ -101,8 +102,34 @@ impl Config {
...
-        let whitelist = self.network.whitelist.iter().map(|s| s.parse().unwrap()).collect::<Vec<_>>();
-        let blacklist = self.network.blacklist.iter().map(|s| s.parse().unwrap()).collect::<Vec<_>>();
+
+        let whitelist = self
+            .network
+            .whitelist_path
+            .as_ref()
+            .map(|path| {
+                fs::read_to_string(path)
+                    .expect("Whitelist file must exist")
```
Comment (kseo) [inline codechain/config/mod.rs]: Not true. The given whitelist file might not exist. Handle the error case by throwing an error.

File: codechain/config/mod.rs [hunk 8]
```diff
@@ -101,8 +102,34 @@ impl Config {
...
+                    .collect()
+            })
+            .unwrap_or_else(Vec::new);
+        let blacklist = self
+            .network
+            .blacklist_path
+            .as_ref()
+            .map(|path| {
+                fs::read_to_string(path)
+                    .expect("Blacklist file must exist")
```
Comment (kseo) [inline codechain/config/mod.rs]: Ditto.

**Comment by jhs7jhs** [comment]
@sgkim126 @kseo Fixed all requested changes.

**Pull request merged**